src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/CodeActionEndpointTest.cs

```diff
@@ -352,6 +360,87 @@ public async Task Handle_MultipleMixedProvider_SupportsCodeActionResolveFalse()
                 c => Assert.True(c.IsCommand));
         }
 
+        [Fact]
+        public async Task GenerateRazorCodeActionContextAsync_WithSelectionRange()
+        {
+            // Arrange
+            var documentPath = "C:/path/to/Page.razor";
+            var codeDocument = CreateCodeDocument("@code {}");
+            var documentResolver = CreateDocumentResolver(documentPath, codeDocument);
+            var codeActionEndpoint = new CodeActionEndpoint(
+                DocumentMappingService,
+                new RazorCodeActionProvider[] {
+                    new MockCodeActionProvider()
+                },
+                Array.Empty<CSharpCodeActionProvider>(),
+                Dispatcher,
+                documentResolver,
+                LanguageServer,
+                LanguageServerFeatureOptions)
+            {
+                _supportsCodeActionResolve = false
+            };
+
+            var initialRange = new Range(new Position(0, 1), new Position(0, 1));
+            var selectionRange = new Range(new Position(0, 5), new Position(0, 5));
+            var request = new RazorCodeActionParams()
+            {
+                TextDocument = new TextDocumentIdentifier(new Uri(documentPath)),
+                Range = initialRange,
+                Context = new ExtendedCodeActionContext()
+                {
+                    SelectionRange = selectionRange,
+                }
+            };
+
+            // Act
+            var razorCodeActionContext = await codeActionEndpoint.GenerateRazorCodeActionContextAsync(request, default);
+
+            // Assert
+            Assert.NotNull(razorCodeActionContext);
+            Assert.Equal(selectionRange, razorCodeActionContext.Request.Range);
+        }
+
+        [Fact]
+        public async Task GenerateRazorCodeActionContextAsync_WithoutSelectionRange()
```

Comment:
WithoutSelectionRange is the default for all other tests isn't it? If that's the case does this actually test anything new?

Reply:
> WithoutSelectionRange is the default for all other tests isn't it? 

Yes

> If that's the case does this actually test anything new?
These two new tests, test `GenerateRazorCodeActionContextAsync`, the others test `Handle`. These also have range specific asserts.